lib/std/os/windows.zig

```diff
@@ -1569,6 +1579,48 @@ pub fn sliceToPrefixedFileW(s: []const u8) !PathSpace {
     return path_space;
 }
 
+/// Convert `dos_path` to an NT path that can be passed to Nt* functions like NtCreateFile.
+pub const NtPath = struct {
+    str: UNICODE_STRING,
+    pub fn init(dos_path: [*:0]const u16) !NtPath {
+        var str : UNICODE_STRING = undefined;
+        const status = ntdll.RtlDosPathNameToNtPathName_U_WithStatus(dos_path, &str, null, null);
+        if (status != .SUCCESS) return unexpectedStatus(status);
```

Comment:
This introduces unnecessary heap allocation, causing `OutOfMemory` to be added to the `OpenError` set. We can do better. Check the wine implementation of `RtlDosPathNameToNtPathName_U_WithStatus` for inspiration. Opening a file should not do heap allocation for the path name.